ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -436,6 +436,122 @@ object Trees {
     val tpe = AnyType
   }
 
+  /** Calls a method inherited from the parent class of `cls` on `receiver`.
+   *
+   *  `cls` must be a Scala.js-defined JS class.
+   *
+   *  Given the Scala.js-defined JS class
+   *
+   *  {{{
+   *  @ScalaJSDefined
+   *  class Foo extends Bar
+   *  }}}
+   *
+   *  The node
+   *
+   *  {{{
+   *  JSBrackerSuperCall(ClassType(Foo), receiver, method, args)
+   *  }}}
+   *
+   *  which is printed as
+   *
+   *  {{{
+   *  receiver.Foo::super[method](...args)
+   *  }}}
+   *
+   *  has the following semantics:
+   *
+   *  {{{
+   *  Bar.prototype[method].call(receiver, ...args)
+   *  }}}
+   *
+   *  If this happens to be located in an instance method of `Foo`, *and*
+   *  `receiver` happens to be `This()`, this is equivalent to the ES6
+   *  statement
+   *
+   *  {{{
+   *  super[method](...args)
+   *  }}}
+   */
+  case class JSBracketSuperCall(cls: ClassType, receiver: Tree, method: Tree,
+      args: List[Tree])(implicit val pos: Position) extends Tree {
+    val tpe = AnyType
+  }
+
+  /** Super constructor call in the constructor of a Scala.js-defined JS class.
+   *
+   *  Exactly one such node must appear in the constructor of a
+   *  Scala.js-defined JS class, at the top-level (possibly as a direct child
+   *  of a top-level `Block`).
+   *
+   *  Statements before this node, as well as the `args`, cannot contain any
+   *  `This()` node. Statements after this node can use `This()`.
+   *
+   *  This node's semantics are to call the super constructor, then create
+   *  the fields of the object declared by the current class, initialized to
+   *  the zero of their type.
```

Comment:
While I see that this is probably a good abstraction for "object initialization", I think the name is really poor. While it *does* call the super constructor, it is by far not everything it does...

How about: `JSBasicClassInitialization`?